From 10072a061310ba950ce95fc459229c1fafe23f56 Mon Sep 17 00:00:00 2001
From: Dong Aisheng <b29396@freescale.com>
Date: Tue, 17 Sep 2013 16:25:44 +0800
Subject: [PATCH 0311/1072] ARM: dts: imx6sl: add pinctrl uhs states for usdhc

git://git.freescale.com/imx/linux-2.6-imx.git imx_3.10.17_1.0.0_beta
commit 0e9cd9dc6f68a550a155181c6f65454dd61419d9

This is needed for SD3.0 cards working on UHS mode.

Signed-off-by: Dong Aisheng <b29396@freescale.com>
Signed-off-by: Shawn Guo <shawn.guo@linaro.org>
Signed-off-by: Hongbo Zhong <hongbo.zhong@windriver.com>
---
 arch/arm/boot/dts/imx6sl-evk.dts |   12 ++++-
 arch/arm/boot/dts/imx6sl.dtsi    |   76 ++++++++++++++++++++++++++++++++++++++
 2 files changed, 85 insertions(+), 3 deletions(-)

diff --git a/arch/arm/boot/dts/imx6sl-evk.dts b/arch/arm/boot/dts/imx6sl-evk.dts
index 2ed81d1..9a89087 100644
--- a/arch/arm/boot/dts/imx6sl-evk.dts
+++ b/arch/arm/boot/dts/imx6sl-evk.dts
@@ -573,8 +573,10 @@
 };
 
 &usdhc1 {
-	pinctrl-names = "default";
+	pinctrl-names = "default", "state_100mhz", "state_200mhz";
 	pinctrl-0 = <&pinctrl_usdhc1_1>;
+	pinctrl-1 = <&pinctrl_usdhc1_1_100mhz>;
+	pinctrl-2 = <&pinctrl_usdhc1_1_200mhz>;
 	bus-width = <8>;
 	cd-gpios = <&gpio4 7 0>;
 	wp-gpios = <&gpio4 6 0>;
@@ -582,16 +584,20 @@
 };
 
 &usdhc2 {
-	pinctrl-names = "default";
+	pinctrl-names = "default", "state_100mhz", "state_200mhz";
 	pinctrl-0 = <&pinctrl_usdhc2_1>;
+	pinctrl-1 = <&pinctrl_usdhc2_1_100mhz>;
+	pinctrl-2 = <&pinctrl_usdhc2_1_200mhz>;
 	cd-gpios = <&gpio5 0 0>;
 	wp-gpios = <&gpio4 29 0>;
 	status = "okay";
 };
 
 &usdhc3 {
-	pinctrl-names = "default";
+	pinctrl-names = "default", "state_100mhz", "state_200mhz";
 	pinctrl-0 = <&pinctrl_usdhc3_1>;
+	pinctrl-1 = <&pinctrl_usdhc3_1_100mhz>;
+	pinctrl-2 = <&pinctrl_usdhc3_1_200mhz>;
 	cd-gpios = <&gpio3 22 0>;
 	status = "okay";
 };
diff --git a/arch/arm/boot/dts/imx6sl.dtsi b/arch/arm/boot/dts/imx6sl.dtsi
index f59bd26..ed78f03 100644
--- a/arch/arm/boot/dts/imx6sl.dtsi
+++ b/arch/arm/boot/dts/imx6sl.dtsi
@@ -1162,6 +1162,37 @@
 				MX6SL_PAD_SD1_DAT7__SD1_DATA7 0x17059
 			>;
 		};
+
+		pinctrl_usdhc1_1_100mhz: usdhc1grp-1-100mhz {
+			fsl,pins = <
+				MX6SL_PAD_SD1_CMD__SD1_CMD 0x170b9
+				MX6SL_PAD_SD1_CLK__SD1_CLK 0x100b9
+				MX6SL_PAD_SD1_DAT0__SD1_DATA0 0x170b9
+				MX6SL_PAD_SD1_DAT1__SD1_DATA1 0x170b9
+				MX6SL_PAD_SD1_DAT2__SD1_DATA2 0x170b9
+				MX6SL_PAD_SD1_DAT3__SD1_DATA3 0x170b9
+				MX6SL_PAD_SD1_DAT4__SD1_DATA4 0x170b9
+				MX6SL_PAD_SD1_DAT5__SD1_DATA5 0x170b9
+				MX6SL_PAD_SD1_DAT6__SD1_DATA6 0x170b9
+				MX6SL_PAD_SD1_DAT7__SD1_DATA7 0x170b9
+			>;
+		};
+
+		pinctrl_usdhc1_1_200mhz: usdhc1grp-1-200mhz {
+			fsl,pins = <
+				MX6SL_PAD_SD1_CMD__SD1_CMD 0x170f9
+				MX6SL_PAD_SD1_CLK__SD1_CLK 0x100f9
+				MX6SL_PAD_SD1_DAT0__SD1_DATA0 0x170f9
+				MX6SL_PAD_SD1_DAT1__SD1_DATA1 0x170f9
+				MX6SL_PAD_SD1_DAT2__SD1_DATA2 0x170f9
+				MX6SL_PAD_SD1_DAT3__SD1_DATA3 0x170f9
+				MX6SL_PAD_SD1_DAT4__SD1_DATA4 0x170f9
+				MX6SL_PAD_SD1_DAT5__SD1_DATA5 0x170f9
+				MX6SL_PAD_SD1_DAT6__SD1_DATA6 0x170f9
+				MX6SL_PAD_SD1_DAT7__SD1_DATA7 0x170f9
+			>;
+		};
+
 	};
 
 	usdhc2 {
@@ -1175,6 +1206,28 @@
 				MX6SL_PAD_SD2_DAT3__SD2_DATA3 0x17059
 			>;
 		};
+
+		pinctrl_usdhc2_1_100mhz: usdhc2grp-1-100mhz {
+			fsl,pins = <
+				MX6SL_PAD_SD2_CMD__SD2_CMD    0x170b9
+				MX6SL_PAD_SD2_CLK__SD2_CLK    0x100b9
+				MX6SL_PAD_SD2_DAT0__SD2_DATA0 0x170b9
+				MX6SL_PAD_SD2_DAT1__SD2_DATA1 0x170b9
+				MX6SL_PAD_SD2_DAT2__SD2_DATA2 0x170b9
+				MX6SL_PAD_SD2_DAT3__SD2_DATA3 0x170b9
+			>;
+		};
+
+		pinctrl_usdhc2_1_200mhz: usdhc2grp-1-200mhz {
+			fsl,pins = <
+				MX6SL_PAD_SD2_CMD__SD2_CMD    0x170f9
+				MX6SL_PAD_SD2_CLK__SD2_CLK    0x100f9
+				MX6SL_PAD_SD2_DAT0__SD2_DATA0 0x170f9
+				MX6SL_PAD_SD2_DAT1__SD2_DATA1 0x170f9
+				MX6SL_PAD_SD2_DAT2__SD2_DATA2 0x170f9
+				MX6SL_PAD_SD2_DAT3__SD2_DATA3 0x170f9
+			>;
+		};
 	};
 
 	usdhc3 {
@@ -1188,5 +1241,28 @@
 				MX6SL_PAD_SD3_DAT3__SD3_DATA3 0x17059
 			>;
 		};
+
+		pinctrl_usdhc3_1_100mhz: usdhc3grp-1-100mhz {
+			fsl,pins = <
+				MX6SL_PAD_SD3_CMD__SD3_CMD    0x170b9
+				MX6SL_PAD_SD3_CLK__SD3_CLK    0x100b9
+				MX6SL_PAD_SD3_DAT0__SD3_DATA0 0x170b9
+				MX6SL_PAD_SD3_DAT1__SD3_DATA1 0x170b9
+				MX6SL_PAD_SD3_DAT2__SD3_DATA2 0x170b9
+				MX6SL_PAD_SD3_DAT3__SD3_DATA3 0x170b9
+			>;
+		};
+
+		pinctrl_usdhc3_1_200mhz: usdhc3grp-1-200mhz {
+			fsl,pins = <
+				MX6SL_PAD_SD3_CMD__SD3_CMD    0x170f9
+				MX6SL_PAD_SD3_CLK__SD3_CLK    0x100f9
+				MX6SL_PAD_SD3_DAT0__SD3_DATA0 0x170f9
+				MX6SL_PAD_SD3_DAT1__SD3_DATA1 0x170f9
+				MX6SL_PAD_SD3_DAT2__SD3_DATA2 0x170f9
+				MX6SL_PAD_SD3_DAT3__SD3_DATA3 0x170f9
+			>;
+		};
+
 	};
 };
-- 
1.7.5.4

